pkgs/by-name/ti/tinyxxd/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tinyxxd";
+  version = "1.3.3";
+  src = fetchFromGitHub {
+    repo = "tinyxxd";
+    owner = "xyproto";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-mup4xHt5Prpcwz2QU0qz+9/KYOE0HNrLx9b7z5o47ag=";
+  };
+
+  outputs = [
+    "out"
+    "xxd"
+  ];
+
+  installFlags = [ "PREFIX=$(out)" ];
+
+  postInstall = ''
+    mkdir -p $xxd/bin
+    ln -s $out/bin/tinyxxd $xxd/bin/xxd
+  '';
```

Comment:
I really do think that the additional output here adds no value; the only reason to separate out aliases like this is if you’re worried about polluting a shared global namespace, which isn’t a concern for Nix. I’d prefer to match what the upstream‐maintainer‐maintained Arch package does here and just `ln -s` directly into `$out`. (Who is going to think “I really want `tinyxxd` but not, like, as `xxd`”? I struggle to imagine the guy. Having `pkgs.tinyxxd` as distinct from `pkgs.xxd` but actually they’re the same – it just complicates the story, in my opinion. The core function of this piece of software is to provide an `xxd` you can use.)

If you have an intense strength of feeling about this, though, then I’ll leave it at this grumble.

Reply:
I like that, as currently modeled, if you ask for `tinyxxd`, you get `tinyxxd`, and if you ask for `xxd`, you get something that fulfills the role of `xxd`. The symmetry with `vim.xxd` is also (IMO) great.

I agree that basically nobody is going to install `tinyxxd`, and that everybody will use the redirector in `unixtools`. _That's_ where the logic that you're talking about -- I want an xxd -- really holds.

I'll stew on it. Still in the boat of keeping it as is though.

Reply:
Multiple outputs can complicate things. We should just use the $out output here unless we have a good reason that this isn't enough.